File: TypeScript/ExampleTests/production/Match.ts
import { AddressComponents } from "./AddressComponents.js";

export class Match {
  matchedAddress: string;
  addressComponents: AddressComponents;

  constructor(matchedAddress: string, addressComponents: AddressComponents) {
    this.matchedAddress = matchedAddress;
    this.addressComponents = addressComponents;
  }
}

File: Rust/ExampleTests/production/match_.rs
use super::*;

#[derive(Debug, Clone, PartialEq)]
pub struct Match {
    pub matchedaddress: String,
    pub addresscomponents: AddressComponents,
}

impl Match {
    pub fn new(matchedaddress: String, addresscomponents: AddressComponents) -> Self {
        Self { matchedaddress,  addresscomponents }
    }
}
